### PR TITLE
Add Docs regarding writing Unit and Integration Testing

### LINKED ISSUE #3553: [Docs] Add Docs regarding writing Unit and Integration Testing
#### Desired State
Add Docs for contributing in Mesheryctl Unit Testing and Integration Testing to the project.
Link to the document- https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.rzpmb66db1sq

---

#### Contributor Resources

- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/meshery/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
- [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ishitakeshawani**
Signed-off-by: ishitakeshawani <keshwaniishita054@gmail.com>

**Description**

This PR fixes #3553

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 6061eb988a74adf6f65073ceb8a30687f0cd09ae at: https://60f47eeeb66d4bea4903c7fb--meshery-docs-preview.netlify.app

**Review by alphaX86 — DISMISSED** [review]
Reviewed the page and output. LGTM on my side 🎉

**Review by piyushsingariya — CHANGES_REQUESTED** [review]

File: docs/pages/project/contributing/contributing-cli.md [hunk 1]
```diff
@@ -128,6 +128,76 @@ When designing for the command line interface, ask and consider the following qu
...
+- [Unit testing CLIs in Go](https://medium.com/swlh/unit-testing-cli-programs-in-go-6275c85af2e7)
+- [How to test CLI commands with Go and Cobra](https://gianarb.it/blog/golang-mockmania-cli-command-with-cobra)
+
+### Key principles
+
+The following key principles should be taken to mind when writing tests:
+1. Golang’s standard library will be used to write tests.
+2. The tests should cover all possible use cases and not just the happy paths.
+3. Integration tests should contain the keyword **“Integration”** in the title and should be marked to be skipped under unit testing. (See below)
+4. **Fixtures** are the mock responses of an API call.
```
[the remark this answers is not in the document]
Comment (ishitakeshawani) [reply docs/pages/project/contributing/contributing-cli.md]: Done

**Comment by l5io** [comment]
🚀 Preview for commit 64285682faf5e52b6dd3985826146e3eeb49d743 at: https://60f51b0d72767e8eb0034b1b--meshery-docs-preview.netlify.app

**Review by adithyaakrishna — APPROVED** [review]
LGTM 🎉 
Checked the Preview as well as links on the Docs 💯

**Pull request merged**